ggml/src/ggml-vulkan/ggml-vulkan.cpp

```diff
@@ -7528,6 +7555,37 @@ static void ggml_vk_timestep_embedding(ggml_backend_vk_context * ctx, vk_context
     }, dryrun);
 }
 
+static void ggml_vk_conv_transpose_1d(ggml_backend_vk_context * ctx, vk_context& subctx, const ggml_tensor * src0, const ggml_tensor * src1, ggml_tensor * dst, bool dryrun = false) {
+    // src0: (K, Cout, Cin, 1) -- kernel
+    // src1: (L, Cin, 1, 1) -- input
+    // dst: (*, Cout, 1, 1)
+
+    GGML_ASSERT(src0->type == GGML_TYPE_F32);
+    GGML_ASSERT(src1->type == GGML_TYPE_F32);
+    GGML_ASSERT( dst->type == GGML_TYPE_F32);
+
+    GGML_TENSOR_BINARY_OP_LOCALS
+
+    GGML_ASSERT(nb00 == sizeof(float));
+    GGML_ASSERT(nb10 == sizeof(float));
+
+    const int32_t s0 = dst->op_params[0];
+
+    vk_op_conv_transpose_1d_push_constants p{};
+    p.Cout = ne01;
```

Comment:
You may get warnings about assigning int64 to int32, particularly on gcc/clang. I'll trigger the CI in a moment which may show these.

Reply:
> You may get warnings about assigning int64 to int32, particularly on gcc/clang. I'll trigger the CI in a moment which may show these.

That's interesting because I use gcc for development and did not get any warnings. Furthermore, this misuse is common in ggml-vulkan.cpp, e.g. `channels` field of `vk_op_conv2d_dw_push_constants` is also uint32_t and int64_t is assigned.

Reply:
Added casts in c56820d30cd7ef393a239af1ec55a44b2a7c635a.